src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -114,13 +118,35 @@ newtype PlonkWitnessMap d = PlonkWitnessMap (Map.Map Natural F -> (PolyVec F d,
 
 newtype PlonkWitnessInput = PlonkWitnessInput (Map.Map Natural F)
 -- TODO (Issue #25): make a proper implementation of Show
-instance Show PlonkWitnessInput where
-    show _ = "PlonkWitnessInput"
--- TODO (Issue #25): make a proper implementation of Arbitrary
 instance Arbitrary PlonkWitnessInput where
     arbitrary = do
-        x <- arbitrary
-        return $ PlonkWitnessInput $ fromList [(1, x), (2, 15//x)]
+        keysLen <- ceiling @Double . logBase 2.0 . fromIntegral <$> chooseInteger (0, 100)
+        let keys = [0..keysLen]
+        values <- vector (integerToInt $ naturalToInteger keysLen) :: Gen [F]
+        let wi = Map.fromList $ zip keys values
+        return (PlonkWitnessInput wi)
+
+instance Show PlonkWitnessInput where
+    show (PlonkWitnessInput m) = "Witness Input: " ++ show m
+
+data (Arithmetic a, KnownNat n) => ACandWitness n a = ACandWitness
```

Comment:
I suggest renaming it to `ArithmeticCircuitTest`, using `Map` instead of `PlonkWitnessInput`, and moving this definition to the `ArithmeticCircuit` module.